### ISSUE TITLE
signInWithCustomToken: type 'Null' is not a subtype of type 'String' in type cast

### ISSUE THREAD

**Issue by pulyaevskiy** [Skycoder42/firebase_auth_rest#5]
```
[ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: type 'Null' is not a subtype of type 'String' in type cast
#0      _$_$CustomTokenSignInResponseFromJson (package:firebase_auth_rest/src/models/signin_response.g.dart:106:30)
#1      new _$CustomTokenSignInResponse.fromJson (package:firebase_auth_rest/src/models/signin_response.freezed.dart:1703:7)
#2      RestApi.signInWithCustomToken (package:firebase_auth_rest/src/rest_api.dart:99:33)
<asynchronous suspension>
#3      FirebaseAuth.signInWithCustomToken (package:firebase_auth_rest/src/firebase_auth.dart:209:9)
<asynchronous suspension>
```

Which is here:

```dart
_$CustomTokenSignInResponse _$_$CustomTokenSignInResponseFromJson(
    Map<String, dynamic> json) {
  return _$CustomTokenSignInResponse(
    idToken: json['idToken'] as String,
    refreshToken: json['refreshToken'] as String,
    expiresIn: json['expiresIn'] as String,
    localId: json['localId'] as String, // <<<< NPE
  );
}
```

In the code `CustomTokenSignInResponse` declares `localId`, however looking at the doc it should not be there:
https://firebase.google.com/docs/reference/rest/auth#section-verify-custom-token

![image](https://user-images.githubusercontent.com/441756/140254816-c36fa9e4-fd9c-434d-ba16-e9c1554afff7.png)

**Comment by Skycoder42** [comment]
I think this is because I falsely assumed it exists on any auth response. I will have to adjust the models. I will try to do so without breaking compability

**Comment by Skycoder42** [comment]
Should be fixed with version 2.0.2. Once the pipelines have run successfully, it should be available within a few hours
